stdlib/REPL/src/REPL.jl

```diff
@@ -1230,22 +1230,35 @@ function setup_interface(
             if isempty(s) || position(LineEdit.buffer(s)) == 0
                 global pkg_mode
                 if pkg_mode === nothing
-                    pkgid = Base.PkgId(Base.UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f"), "Pkg")
-                    REPLExt = Base.require_stdlib(pkgid, "REPLExt")
-                    pkg_mode = nothing
-                    if REPLExt isa Module && isdefined(REPLExt, :PkgCompletionProvider)
-                        for mode in repl.interface.modes
-                            if mode isa LineEdit.Prompt && mode.complete isa REPLExt.PkgCompletionProvider
-                                pkg_mode = mode
-                                break
+                    LineEdit.clear_line(LineEdit.terminal(s))
+                    Base.printstyled(LineEdit.terminal(s), "(...) pkg> ", color = :blue)
+                    # do the mode switch async but spawn Pkg load to avoid blocking typing during loading
+                    t_pkgload = Threads.@spawn begin
+                        pkgid = Base.PkgId(Base.UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f"), "Pkg")
+                        Base.require_stdlib(pkgid, "REPLExt")
+                    end
+                    t_replswitch = @async begin
```

Comment:
Why not `@spawn`?

Reply:
I had an earlier design that required two tasks, but I think you're right, and this can be simplified into a single spawn now

Reply:
Even still, `@async` should probably never be used because it does bad things

Reply:
What if you want a task that strictly stays on the same thread as the task it was created from? Is that just a bad pattern?

Reply:
Yep, because threads aren't really concept that exist in julia, so it's basically an implementation detail.

Reply:
I think you mean `@async`, but thanks for the explanation. 

Btw, maybe we should just go ahead and deprecate the Threads module to Tasks. Perhaps without a warning though, so just the alias and a docs note.

Reply:
Yeah, that name isn't very helpful.